File: 1002-find-common-characters/1002-find-common-characters.java
class Solution {
    public List<String> commonChars(String[] words) {
        int[] alCnt = new int[27];
        int[] tmpCnt = new int[27];
        
        for(int i = 0; i < words[0].length(); i++) {
            alCnt[words[0].charAt(i) - 97]++;
        }
        
        for(String word : words) {
            for(int i = 0; i < word.length(); i++) {
                tmpCnt[word.charAt(i) - 97]++;
            }
            for(int i = 0; i < 27; i++) {
                alCnt[i] = Math.min(alCnt[i], tmpCnt[i]);
                tmpCnt[i] = 0;
            }
        }
        
        List<String> answer = new ArrayList<>();
        String c;
        for(int i = 0; i < 27; i++) {
            c = (char)(i + 97) + "";
            for(int j = 0; j < alCnt[i]; j++) {
                answer.add(c);
            }
        }
        
        return answer;
    }
}

File: 1002-find-common-characters/1002-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        alCnt = [words[0].count(chr(i + 97)) for i in range(27)]
        
        for word in words:
            for n in range(27):
                alCnt[n] = min(alCnt[n], word.count(chr(n + 97)))
                
        return [chr(i + 97) for i, cnt in enumerate(alCnt) for j in range(cnt) if cnt > 0]
        
        
        